tests/boards/tenstorrent/tt_blackhole/clock_control/src/main.c

```diff
@@ -33,52 +33,46 @@ ZTEST(clock_control_rate, test_set_rate_gddr)
 {
 	const struct device *pll = DEVICE_DT_GET(DT_NODELABEL(pll0));
 	uint32_t new_rate;
-	const uint32_t target_rate = 700;
 	int ret;
 
 	clock_control_subsys_t gddr_subsys =
 		(clock_control_subsys_t)CLOCK_CONTROL_TT_BH_CLOCK_GDDRMEMCLK;
 
-	ret = clock_control_set_rate(pll, gddr_subsys, (clock_control_subsys_rate_t)target_rate);
-	zassert_ok(ret, "set_rate for GDDR failed with %d", ret);
+	for (uint32_t target_rate = 750; target_rate <= 1000; target_rate += 50) {
+		ret = clock_control_set_rate(pll, gddr_subsys,
+					     (clock_control_subsys_rate_t)target_rate);
+		zassert_ok(ret, "set_rate for GDDR failed with %d", ret);
 
-	ret = clock_control_get_rate(pll, gddr_subsys, &new_rate);
-	zassert_ok(ret, "get_rate for GDDR failed with %d", ret);
+		ret = clock_control_get_rate(pll, gddr_subsys, &new_rate);
+		zassert_ok(ret, "get_rate for GDDR failed with %d", ret);
 
-	zassert_within(new_rate, target_rate, new_rate / 100.f * CLOCK_RATE_TOLERANCE_PERCENT,
-		       "Expected ~%d Hz but got %d Hz", target_rate, new_rate);
+		zassert_within(new_rate, target_rate,
+			       new_rate / 100.f * CLOCK_RATE_TOLERANCE_PERCENT,
+			       "Expected ~%d Hz but got %d Hz", target_rate, new_rate);
+	}
 }
 
 ZTEST(clock_control_rate, test_set_rate_aiclk)
 {
 	const struct device *pll = DEVICE_DT_GET(DT_NODELABEL(pll0));
 	uint32_t new_rate;
-	const uint32_t target_rate = 750;
 	int ret;
 
 	clock_control_subsys_t aiclk_subsys =
 		(clock_control_subsys_t)CLOCK_CONTROL_TT_BH_CLOCK_AICLK;
 
-	ret = clock_control_set_rate(pll, aiclk_subsys, (clock_control_subsys_rate_t)target_rate);
-	zassert_ok(ret, "set_rate for AICLK failed with %d", ret);
-
-	ret = clock_control_get_rate(pll, aiclk_subsys, &new_rate);
-	zassert_ok(ret, "get_rate for AICLK failed with %d", ret);
+	for (uint32_t target_rate = 200; target_rate <= 1350; target_rate += 50) {
```

Comment:
Same - would be good to use Kconfig for min, max, and increment.

 I guess if there are multiple domains, then they would need to have slightly different Kconfig variable names.